test/node-next/es2018/cursor_async_iterator.test.js

```diff
@@ -72,16 +72,17 @@ describe('Cursor Async Iterator Tests', function () {
     }
   });
 
-  it('should properly stop when cursor is closed', {
+  // TODO - figure out how to get this test working again
```

Comment:
The `for await ... of`  statement consumes the entire stream before entering the body of the loop, so this test fails with `count` larger than 1. I'm not sure if that means there's something incorrect with the implementation, or if this is a bad test.

Reply:
@mbroadst This test is still failing if un-skipped, any thoughts on what could be going wrong here?

Reply:
There are a couple of issues here:
- We removed the [Symbol.asyncIterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of) implementation for cursor because it already existed on `Readable`, so that actually needs to be reimplemented here because we want people to be able to do `for await (const doc of cursor) { ... }`

- If this is currently broken, that means that `CursorStream` is broken somehow. You might just have a silly bug here where you're creating a new stream on every iteration, in which case this fix should suffice:
```
const cursorStream = cursor.stream();
for await (const doc of cursorStream) { ... }
```
Otherwise, this would indicate that when you close the cursor it is not closing the stream which might be just fine! We assume that the stream might consume a bunch of document on that first iteration so this is probably desired behavior.

Let's fix the first problem, and revert the changes to the test here as your validation that the async iterator is implemented properly.